README.md

```diff
@@ -1,10 +1,260 @@
+bind-mounted from the developer's repository. Tests may launch Nginx multiple
+times, in different configuration, depending on what is needed. After the
+tests runner finishes, all the processes and the environment is cleaned up
+by pytest.
+
+Below, there is a general overview of all the components of the test harness.
+More detailed documentation can be found in docstrings and comments in the code
+itself. If in doubt, it also may be very helpful to look for examples among
+existing tests.
+
+### Quickstart
+In order to execute all the test, just issue:
+
+	make test
+
+In order have fine grained control over pytest command, execute:
+
+	make shell
+
+This will launch an interactive environment inside the container that has all
+the dependencies installed. The developer may now launch the `pytest` command
+the way it's necessary, debug the environment and temporarily add/change
+the dependencies.
+
+### Makefile
+Makefile provides an easy way to start the testing environment without the need
+to worry about the correct docker commands. It's core concept is `adminrouter-devkit`
+container which is contains all the dependencies that are needed to run
+Adminrouter, and inside which all tests-related commands are run.
+
+It exposes a couple of targets:
+* `make clean` - remove all containers created by the test harness. It does not
+  remove the images themselves though as the layer cache may be useful later
+  on and the user may remove them themselves.
+* `make devkit` - creates the `adminrouter-devkit` container. By default other
+   targets execute this step automatically if devkit container image does not
+   exist yet.
+* `make update-devkit` - updates `adminrouter-devkit`. Should be run every time
+   the Dockerfile or its dependencies change.
+* `make tests` - launch all the tests. Worth noting is the fact that mccabe
+  complexity of the code is also verified and an error is raised if it's equal
+  or above 10.
+* `make shell` - launch an interactive shell within the devkit container. Should
+  be used in case when fine grained control of the tests is necessary or
+  during debugging.
+* `make flake8` - launch flake8 which will check all the tests and test-harness
+  files by default.
+
+### Docker container
+As mentioned earlier, all the commands are executed inside the `adminrouter-devkit`
+container`. It follows the same build process for Nginx that happens during
+DC/OS build with the execption of setting the  `--with-debug` flag. It also
+contains some basic debugging tools, py.test related dependencies and basic
+files that help py.test mimic the DC/OS environment. Their location is then
+exported via environment variables to the py.test code, so changing their location
+can be done by edition only the Dockerfile.
+
+Due to some issues with overlayfs on older kernels, the container links
+/tmp directory to /dev/shm.
+
+### Repository flavours
+Adminrouter reposity can come in two variants/flavours - Enterprise and Opensource.
```

Comment:
yep, I did. I missed that one though - VIM spell-checking just highlights words, and if you have too many domain-specific ones in the text, it's sometimes difficult to spot legitimate mistakes.